ludwig/data/dataframe/dask.py

```diff
@@ -0,0 +1,109 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2020 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import multiprocessing
+import os
+
+import dask
+import dask.array as da
+import dask.dataframe as dd
+
+from ludwig.constants import NAME, PROC_COLUMN
+from ludwig.data.dataset.parquet import ParquetDataset
+from ludwig.data.dataframe.base import DataFrameEngine
+from ludwig.utils.data_utils import DATA_PROCESSED_CACHE_DIR, DATASET_SPLIT_URL
+from ludwig.utils.misc_utils import get_combined_features
+
+TMP_COLUMN = '__TMP_COLUMN__'
+
+
+def set_scheduler(scheduler):
+    dask.config.set(scheduler=scheduler)
+
+
+class DaskEngine(DataFrameEngine):
+    def __init__(self, parallelism=None):
+        self._parallelism = parallelism or multiprocessing.cpu_count()
+
+    def empty_df_like(self, df):
+        # Our goal is to preserve the index of the input dataframe but to drop
+        # all its columns. Because to_frame() creates a column from the index,
+        # we need to drop it immediately following creation.
+        return df.index.to_frame(name=TMP_COLUMN).drop(columns=[TMP_COLUMN])
```

Comment:
@clarkzinzow let me know if there's a better way to do this

Reply:
Would something like `pd.DataFrame(index=df.index)` work? Assuming that this is supposed to return a Pandas dataframe, if it's supposed to return a Dask dataframe then passing that to `dd.from_pandas()` would do the trick.

Reply:
Both the input and output dataframes are Dask in this case.  It's an interesting situation because we have a "source" DF and a "sink" in our processing.  We take the columns from the source, transform them, and they become the columns of the sink (one-to-many, so one source column may become multiple columns in the sink DF).  That's why we create this empty DF, but in order to align the columns correctly, we need to preserve the index (hence this hacky looking tmp column).

Reply:
Cool, got it. If you're hoping to avoid the grossness of creating and immediately dropping a temporary column, it seems as if `dd.from_pandas(pd.DataFrame(index=df.index))` would do the trick.

Reply:
Oh interesting, thanks I'll give that a shot!